File: frontend/src/app/shared/model/authentication/authentication-modify-request.ts
export class AuthenticationModifyRequest {
    public readonly newEmail: string;
    public readonly oldEmail: string;
    public readonly password: string;

    constructor(oldEmail: string, newEmail: string, password: string) {
        this.oldEmail = oldEmail;
        this.password = password;
        this.newEmail = newEmail;
    }
}

File: backend/src/main/java/hu/szakdoga/backend/authentication/data/model/AuthenticationModifyRequest.java
package hu.szakdoga.backend.authentication.data.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class AuthenticationModifyRequest {
    private String oldEmail;
    private String newEmail;
    private String password;
}
